### PR TITLE
feat(TextInput): add copyable prop

### PR DESCRIPTION
**Pull request by SamCreasey**
Add a new property to the TextInput component to allow for copying the value of the input. 
Adds a new "useCopy" hook

### CODE REVIEW

File: src/components/TextInput/TextInput.stories.tsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ export default {
         type: TextInputType.Text,
         validation: Validation.Default,
         spellcheck: true,
+        copyable: true,
```
Comment (hochreutenerl) [inline src/components/TextInput/TextInput.stories.tsx:22]: nit: I would suggest false as a default

File: src/components/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -89,6 +94,7 @@ export const TextInput: FC<TextInputProps> = ({
     autocomplete = false,
     dotted = false,
     value = "",
+    copyable,
```
Comment (hochreutenerl) [inline src/components/TextInput/TextInput.tsx:97]: Probably it makes sense to define false as a default value

File: src/components/TextInput/TextInput.tsx [hunk 3]
```diff
@@ -219,6 +229,32 @@ export const TextInput: FC<TextInputProps> = ({
...
+            {copyable && (
+                <button
+                    className={merge([
+                        "tw-flex tw-items-center tw-justify-center tw-transition-colors tw-rounded",
+                        disabled ? "tw-cursor-default tw-text-black-40" : "tw-text-black-60 hover:tw-text-black-100",
+                        copyButtonIsFocusVisible && FOCUS_STYLE,
+                    ])}
+                    onClick={() => copy(value)}
+                    data-test-id="copy-icon"
+                    title="Copy input value"
```
Comment (hochreutenerl) [inline src/components/TextInput/TextInput.tsx:241]: Just for confirmation: Is this the final title?

Comment (SamCreasey) [reply src/components/TextInput/TextInput.tsx:241]: I changed it to "copy input text", but if you have any other ideas I would appreciate it :)

File: src/components/TextInput/TextInput.tsx (hunk 2 above)
Comment (SamCreasey) [reply src/components/TextInput/TextInput.tsx:97]: done

File: src/components/TextInput/TextInput.stories.tsx (hunk 1 above)
Comment (SamCreasey) [reply src/components/TextInput/TextInput.stories.tsx:22]: done

**Review by hochreutenerl — APPROVED** [review]
Nice work 💪

**Pull request merged**